### PR TITLE
Redirect logged out users to sign up page for limited posts

### LINKED ISSUE #3738: the link "show this post" (german: "Beitrag betrachten") in notification-mails shouldn't lead to an 404 error...
..., but should ask the user to sign in and redirect to the post after signing in. up to now you are redirectrd to your stream, which doesn't make sense in my opinion.
also the 404 is "scaring" users a little bit, i think.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/58995-the-link-show-this-post-german-beitrag-betrachten-in-notification-mails-shouldn-t-lead-to-an-404-error?utm_campaign=plugin&utm_content=tracker%2F522&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F522&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by svbergerem**
Fixes #3738.

### CODE REVIEW

**Comment by Flaburgan** [comment]
Once authenticated, will the user be redirected to the post?

**Comment by svbergerem** [comment]
Yes, we basically do the same thing for remote profiles and I also tried it before opening this PR.

**Comment by jhass** [comment]
Merged as 123e6d1dd4f46448f4068fe182c9e2bd2b2636c2

Thanks.

**Comment by Flaburgan** [comment]
Thank you, this was a real UX problem.

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -11,9 +11,13 @@ class PostsController < ApplicationController
...
-    @code = "not-public"
-    respond_to do |format|
-      format.all { render template: "errors/not_public", status: 404, layout: "error_page" }
+    if user_signed_in?
+      @code = "not-public"
+      respond_to do |format|
+        format.all { render template: "errors/not_public", status: 404, layout: "error_page" }
+      end
+    else
+      authenticate_user!
```
Comment (theworldbright) [inline app/controllers/posts_controller.rb]: Good to know this is possible :+1:

**Pull request closed without merging**